extension.js

```diff
@@ -24,173 +24,180 @@ const STATUS_STARTING = _('Initializing');
 const STATUS_CONNECTED = _('Connected');
 const STATUS_DISCONNECTED = _('Disconnected');
 
+const MullvadToggle = GObject.registerClass({
+    GTypeName: 'MullvadToggle',
+}, class MullvadToggle extends QuickSettings.QuickMenuToggle {
+    _init(mullvad) {
+        super._init({
+            label: STATUS_STARTING,
+            gicon: Gio.icon_new_for_string(`${Me.path}/icons/${STATUS_DISCONNECTED}.svg`),
+        });
 
-const MullvadIndicator = GObject.registerClass({
-    GTypeName: 'MullvadIndicator',
-}, class MullvadIndicator extends PanelMenu.SystemIndicator {
+        // Menu section with configurable connection status information
+        this._detailedStatusSection = new PopupMenu.PopupMenuSection();
+        this.menu.addMenuItem(this._detailedStatusSection);
 
-    _init() {
-        super._init(0);
+        // Separator line
+        this.menu.addMenuItem(new PopupMenu.PopupSeparatorMenuItem());
 
-        // Get our settings
-        this._settings = ExtensionUtils.getSettings('org.gnome.Shell.Extensions.MullvadIndicator');
+        // Menu section with refresh and settings buttons
+        this._bottomSection = new PopupMenu.PopupMenuSection();
+        this._buildBottomSection(mullvad);
+        this.menu.addMenuItem(this._bottomSection);
 
-        // Instantiate our Mullvad object
-        this._mullvad = new Mullvad.MullvadVPN();
+        this._sync(mullvad);
+    }
 
-        // Connect our signals
-        this._connectPrefSignals();
-        this._watch = this._mullvad.connect('status-changed', _mullvad => {
-            this._updateGui();
+    _buildBottomSection(mullvad) {
+        // Item for manually checking connection to Mullvad
+        let refreshItem = new PopupMenu.PopupMenuItem(_('Refresh'));
+        refreshItem.actor.connect('button-press-event', () => {
+            mullvad._pollMullvad();
         });
+        this._bottomSection.addMenuItem(refreshItem);
 
-        // Start our GUI and enter our main loop
-        this._initGui();
-        this._main();
+        // Item for opening extension settings
+        let settingsItem = new PopupMenu.PopupMenuItem(_('Settings'));
+        settingsItem.actor.connect('button-press-event', () => {
+            Util.spawnCommandLine('gnome-extensions prefs mullvadindicator@pobega.github.com');
+        });
+        this._bottomSection.addMenuItem(settingsItem);
     }
 
-    _connectPrefSignals() {
```

Comment:
Actually I'm a bit confused. So this PR also guts the ability to control what you see via preferences?

Reply:
This appears to work, sorry for the confusion.